<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>IdeaDatabase</class>
 <widget class="QMainWindow" name="IdeaDatabase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1093</width>
    <height>877</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>IdeaDatabase</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Category Tree View</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QTreeView" name="recordTreeView"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>Selection Field</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="1" column="2">
            <widget class="QLabel" name="updateDateLabel">
             <property name="text">
              <string>Update Date</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Update Date</string>
             </property>
            </widget>
           </item>
           <item row="4" column="2">
            <widget class="QTextEdit" name="appliedSectionTextEdit"/>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Top-Level Category</string>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Mid-Level Category</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="idLabel">
             <property name="text">
              <string>ID Number</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QDateEdit" name="dateAppliedDateEdit"/>
           </item>
           <item row="2" column="2">
            <widget class="QComboBox" name="statusComboBox"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Status</string>
             </property>
            </widget>
           </item>
           <item row="9" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Title</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Applied Section</string>
             </property>
            </widget>
           </item>
           <item row="9" column="2">
            <widget class="QLineEdit" name="titleLineEdit"/>
           </item>
           <item row="11" column="0">
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>Attachment</string>
             </property>
            </widget>
           </item>
           <item row="8" column="2">
            <widget class="QLineEdit" name="lowerLevelLineEdit"/>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>ID</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Date Applied</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Priority</string>
             </property>
            </widget>
           </item>
           <item row="7" column="2">
            <widget class="QComboBox" name="midLevelCategoryComboBox"/>
           </item>
           <item row="10" column="0">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>Description</string>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Lower-Level Category</string>
             </property>
            </widget>
           </item>
           <item row="10" column="2">
            <widget class="QTextEdit" name="descriptionTextEdit"/>
           </item>
           <item row="5" column="2">
            <widget class="QComboBox" name="priorityComboBox"/>
           </item>
           <item row="6" column="2">
            <widget class="QComboBox" name="topLevelCategoryComboBox"/>
           </item>
           <item row="11" column="2">
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <spacer name="horizontalSpacer_6">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="attachmentPushButton">
               <property name="text">
                <string>Attachment</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="attachmentLabel">
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="attachmentDeletePushButton">
               <property name="text">
                <string/>
               </property>
               <property name="icon">
                <iconset resource="IdeaDatabase.qrc">
                 <normaloff>:/images/batsu</normaloff>:/images/batsu</iconset>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_5">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="addPushButton">
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="updatePushButton">
             <property name="text">
              <string>Update</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="clearPushButton">
             <property name="text">
              <string>Clear</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>List View</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,3">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLineEdit" name="searchLineEdit"/>
           </item>
           <item>
            <widget class="QPushButton" name="searchPushButton">
             <property name="text">
              <string>Search</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <item row="1" column="0">
            <widget class="QLabel" name="label_16">
             <property name="text">
              <string>Priority</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QComboBox" name="filterMidLevelCategoryComboBox"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>Top-Level Category</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>Status</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="filterStatusComboBox"/>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_15">
             <property name="text">
              <string>Mid-Level Category</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="filterPriorityComboBox"/>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="filterTopLevelCategoryComboBox"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="deletePushButton">
             <property name="text">
              <string>Delete</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableView" name="recordTableView">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::SelectionMode::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectionBehavior::SelectRows</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1093</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionClose"/>
    <addaction name="actionExport_to_JSON"/>
    <addaction name="actionImport_JSON"/>
    <addaction name="actionSettings"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actionExport_to_JSON">
   <property name="text">
    <string>Export to JSON</string>
   </property>
  </action>
  <action name="actionImport_JSON">
   <property name="text">
    <string>Import JSON</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="IdeaDatabase.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionSettings</sender>
   <signal>triggered(bool)</signal>
   <receiver>IdeaDatabase</receiver>
   <slot>onSettingsClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>546</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>recordTableView</sender>
   <signal>clicked(QModelIndex)</signal>
   <receiver>IdeaDatabase</receiver>
   <slot>onRecordTableViewSelected(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>521</x>
     <y>392</y>
    </hint>
    <hint type="destinationlabel">
     <x>546</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onSettingsClicked()</slot>
  <slot>onRecordTableViewSelected(QModelIndex)</slot>
 </slots>
</ui>
